app/resources/cdl/automatic_approver.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module CDL
+  class AutomaticApprover
+    def self.run
+      change_set_persister.buffer_into_index do |buffered_change_set_persister|
+        new(change_set_persister: buffered_change_set_persister).run
+      end
+    end
+
+    def self.change_set_persister
+      ScannedResourcesController.change_set_persister
+    end
+
+    delegate :metadata_adapter, to: :change_set_persister
+    delegate :query_service, to: :metadata_adapter
+    attr_reader :change_set_persister
+    def initialize(change_set_persister:)
+      @change_set_persister = change_set_persister
+    end
+
+    def run
+      resources = []
+      pending_cdl_resources.each do |pending_resource|
+        next unless ready_for_completion?(pending_resource)
+        change_set = ChangeSet.for(pending_resource)
+        change_set.validate(state: "complete")
+        resource = change_set_persister.save(change_set: change_set)
+        resources += [resource]
+      end
+      notify(resources: resources)
+    end
+
+    def notify(resources:)
+      CDL::CompleteMailer.with(resource_ids: resources.map(&:id).map(&:to_s)).resources_completed.deliver_later
+    end
+
+    def pending_cdl_resources
+      query_service.custom_queries.find_by_property(
+        property: :metadata,
+        value: { state: "pending", change_set: "CDL::Resource" },
+        model: ScannedResource,
+        lazy: true
+      )
+    end
+
+    def ready_for_completion?(resource)
+      CompletionEligibility.new(change_set_persister: change_set_persister, resource: resource).eligible?
+    end
+
+    class CompletionEligibility
+      delegate :metadata_adapter, to: :change_set_persister
+      delegate :query_service, to: :metadata_adapter
+      attr_reader :change_set_persister, :resource
+      def initialize(change_set_persister:, resource:)
+        @change_set_persister = change_set_persister
+        @resource = resource
+      end
+
+      def eligible?
+        all_processed? && pages_match? && manifest_generates?
+      end
+
+      # All child file sets are processed.
+      def all_processed?
+        query_service.custom_queries.find_deep_children_with_property(resource: resource, model: FileSet, property: :processing_status, value: "in process", count: true).zero?
+      end
+
+      def pages_match?
+        first_member.mime_type.include?("application/pdf")
+        pdf_page_count == resource.member_ids.size - 1
```

Comment:
Why are they off by one? is page count zero-indexed? Maybe it would be more clear to add 1 to the value in `pdf_page_count` instead of subtracting here.

Reply:
I don't think it would be - there's one less member size, not one less PDF page size. It's because the first member is the PDF itself.

Reply:
Aha, of course, yes a comment will be helpful thank you.